### PR TITLE
feat(comment):add changyan comment system

### PR DESCRIPTION
**Pull request by lslvxy**
**Contributing rules**

- Fork the repo and create your branch from `canary`. Then be sure to put the `canary` branch as the target for your pull request.
- Please be sure to follow the [contributing guidelines](https://github.com/viosey/hexo-theme-material/blob/master/CONTRIBUTING.md), especially for commit message
- Remove the `Contributing rules` part from this description
- Fill out the other parts from this description

**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bug fix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?** (check one with "x")

- [x] Yes
- [ ] No

____

**Description**

add changyan comment system

____

**Verification steps**

No verification steps.

### CODE REVIEW

File: layout/_partial/comment.ejs [hunk 1]
```diff
@@ -3,6 +3,19 @@
     <%- partial('_widget/duoshuo') %>
 <% } %>
 
+<% if(theme.comment.use === 'changyan') { %>
+    <!-- 使用 changyan -->
+    <div id="disqus-comment">
```
Comment (cubesky) [inline layout/_partial/comment.ejs]: edit this to `changyan-comment`

File: layout/_partial/comment.ejs [hunk 2]
```diff
@@ -3,6 +3,19 @@
...
+        <%- partial('_widget/changyan') %>
+    </div>
+    <style>
+        #disqus-comment{
```
Comment (cubesky) [inline layout/_partial/comment.ejs]: edit this to `changyan-comment`

File: layout/_widget/changyan.ejs [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+<script type="text/javascript">
+(function(){
+var appid = '<%= theme.comment.changyan_appid %>';
+var conf = '<%= theme.comment.changyan_conf %>';
+var width = window.innerWidth || document.documentElement.clientWidth;
+if (width < 960) {
+window.document.write('<script id="changyan_mobile_js" charset="utf-8" type="text/javascript" src="http://changyan.sohu.com/upload/mobile/wap-js/changyan_mobile.js?client_id=' + appid + '&conf=' + conf + '"><\/script>'); } else { var loadJs=function(d,a){var c=document.getElementsByTagName("head")[0]||document.head||document.documentElement;var b=document.createElement("script");b.setAttribute("type","text/javascript");b.setAttribute("charset","UTF-8");b.setAttribute("src",d);if(typeof a==="function"){if(window.attachEvent){b.onreadystatechange=function(){var e=b.readyState;if(e==="loaded"||e==="complete"){b.onreadystatechange=null;a()}}}else{b.onload=a}}c.appendChild(b)};loadJs("http://changyan.sohu.com/upload/changyan.js",function(){window.changyan.api.config({appid:appid,conf:conf})}); } })(); </script>
```
Comment (cubesky) [inline layout/_widget/changyan.ejs]: Please edit `http://` to `https://`

File: layout/_partial/comment.ejs (hunk 1 above)
Comment (lslvxy) [inline layout/_partial/comment.ejs]: 已经修改了,要再发一个pr么

File: layout/_widget/changyan.ejs [hunk 4]
```diff
@@ -6,4 +6,4 @@ var appid = '<%= theme.comment.changyan_appid %>';
...
+window.document.write('<script id="changyan_mobile_js" charset="utf-8" type="text/javascript" src="https://changyan.sohu.com/upload/mobile/wap-js/changyan_mobile.js?client_id=' + appid + '&conf=' + conf + '"><\/script>'); } else { var loadJs=function(d,a){var c=document.getElementsByTagName("head")[0]||document.head||document.documentElement;var b=document.createElement("script");b.setAttribute("type","text/javascript");b.setAttribute("charset","UTF-8");b.setAttribute("src",d);if(typeof a==="function"){if(window.attachEvent){b.onreadystatechange=function(){var e=b.readyState;if(e==="loaded"||e==="complete"){b.onreadystatechange=null;a()}}}else{b.onload=a}}c.appendChild(b)};loadJs("http://changyan.sohu.com/upload/changyan.js",function(){window.changyan.api.config({appid:appid,conf:conf})}); } })(); </script>
```
Comment (cubesky) [inline layout/_widget/changyan.ejs]: This still has `http` protocal url

File: layout/_widget/changyan.ejs (hunk 3 above)
Comment (neoFelhz) [inline layout/_widget/changyan.ejs]: Yep... I missed it when I review it.

**Pull request merged**